meos/src/general/ttext_textfuncs.c

```diff
@@ -234,6 +187,39 @@ datum_upper(Datum value)
   return pg_upper(DatumGetTextP(value));
 }
 
+/**
+ * @brief Convert the text value to uppercase
+ * @note Function adapted from the external function @p upper() in file
+ * @p varlena.c. Notice that @p DEFAULT_COLLATION_OID` is used instead of 
+ * @p PG_GET_COLLATION().
+ */
+Datum
+pg_initcap(text *txt)
+{
+  char *out_string;
+  text *result;
+
+#if MEOS
+  out_string = asc_initcap(VARDATA_ANY(txt), VARSIZE_ANY_EXHDR(txt));
+#else /* ! MEOS */
+  out_string = str_initcap(VARDATA_ANY(txt), VARSIZE_ANY_EXHDR(txt),
+    DEFAULT_COLLATION_OID);
+#endif /* MEOS */
+  result = cstring2text(out_string);
+  pfree(out_string);
+
+  return PointerGetDatum(result);
+}
+
+/**
+ * @brief Convert the text value to uppercase
+ */
+Datum
+datum_initcap(Datum value)
+{
+  return pg_initcap(DatumGetTextP(value));
+}
```

Comment:
Do we need 2 functions?
If so, `pg_initcap` should probably return a `text *` instead of a `Datum`.